File: web/src/core/types/resource.ts
export type ResourceState = 'ACTIVE' | 'READY' | 'PROCESSING' | 'PENDING' | 'FAILED' | string;
export type ResourceScope = 'PRIVATE' | 'PUBLIC';

export interface Category {
  id: string;
  name: string;
  parent_id?: string;
}

export interface CategoryNode extends Category {
  children?: CategoryNode[];
}

export interface ResourceVersion {
  id: string;
  resource_id: string;
  version_num: number;
  semver?: string;
  state: ResourceState;
  file_size: number;
  download_url?: string;
  meta_data?: Record<string, any>;
  created_at: string;
}

export interface Resource {
  id: string;
  name: string;
  type_key: string;
  category_id?: string;
  owner_id: string;
  scope: ResourceScope;
  tags: string[];
  created_at: string;
  updated_at: string;
  latest_version_id?: string;
  latest_version?: ResourceVersion;
}

export interface ResourceDependency {
  id: string;
  resource_id: string;
  resource_name: string;
  type_key?: string;
  version_id: string;
  semver?: string;
  constraint?: string;
  dependencies?: ResourceDependency[];
}

File: sdk/java/src/main/java/io/simhub/sdk/model/Resource.java
package io.simhub.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.util.Date;
import java.util.List;

@Data
public class Resource {
    private String id;
    
    @JsonProperty("type_key")
    private String typeKey;
    
    @JsonProperty("resource_type")
    private ResourceType resourceType;
    
    @JsonProperty("category_id")
    private String categoryId;
    
    private Category category;
    
    private String name;
    
    @JsonProperty("owner_id")
    private String ownerId;
    
    private String scope;
    
    private List<String> tags;
    
    @JsonProperty("latest_version_id")
    private String latestVersionId;
    
    @JsonProperty("latest_version")
    private ResourceVersion latestVersion;
    
    @JsonProperty("created_at")
    private Date createdAt;
    
    @JsonProperty("updated_at")
    private Date updatedAt;
}
